jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2150,25 +2150,22 @@ void java_bytecode_convert_methodt::convert_invoke(
     }
   }
 
-  code_function_callt call;
   location.set_function(method_id);
 
-  call.add_source_location() = location;
-  call.arguments() = pop(parameters.size());
+  code_function_callt::argumentst arguments = pop(parameters.size());
 
   // double-check a bit
-  if(use_this)
-  {
-    const exprt &this_arg = call.arguments().front();
-    INVARIANT(
-      this_arg.type().id() == ID_pointer, "first argument must be a pointer");
-  }
+  INVARIANT(
+    !use_this || arguments.front().type().id() == ID_pointer,
+    "first argument must be a pointer");
 
   // do some type adjustment for the arguments,
   // as Java promotes arguments
   // Also cast pointers since intermediate locals
   // can be void*.
-
+  INVARIANT(
+    parameters.size() <= arguments.size(),
```

Comment:
For Java I'm 99% certain this can be `==`, as Java varargs are syntactic sugar for an Object[] parameter